src/common/opencl.c

```diff
@@ -1109,6 +1110,10 @@ void dt_opencl_cleanup(dt_opencl_t *cl)
                                     "successful and %d events lost, max event=%d\n",
             cl->dev[i].name, i, cl->dev[i].totalsuccess, cl->dev[i].totalevents, cl->dev[i].totallost,
             cl->dev[i].maxeventslot);
+          if(cl->dev[i].maxeventslot >= 1024)
+            fprintf(stderr, "Your OpenCl device '%s' had used up to %d events while processing. This might be safe\n"
+                            "but there is currently no correct check implemented. Take care!\n",
+                             cl->dev[i].name, cl->dev[i].maxeventslot); 
```

Comment:
Sorry to comment late, but today is the first time I saw this message.  During export I routinely use ~3000 events with no problem, so I'm not sure where the threshold of 1024 comes in.  I hit that just importing a few images and opening a couple in darkroom.

But, more importantly if a user gets this message, then they are going to think there is a problem and start filing issues that have to be dealt with.  There is no actionable content.  How does the user "take care"?  

I suggest that you either get rid of this message, or make it a debugging message so that users don't get confused.

Reply:
Yes, a debug option would be better here.

The problem is, we don't know for sure the number of events the device can handle. Most have unrestricted, some allow only 1k. We can test for that, would require CL 2.0